bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java

```diff
@@ -50,25 +55,41 @@
 @NonNullByDefault
 @Component(immediate = true, service = { IconProvider.class })
 public class FreeboxOsIconProvider extends AbstractResourceIconProvider {
+    private static final String DEFAULT_DESCRIPTION = "Icons provided by FreeboxOS iteself";
+    private static final String DEFAULT_LABEL = "FreeboxOS Icons";
+    private static final int REQUEST_TIMEOUT_MS = 8000;
 
     private final Logger logger = LoggerFactory.getLogger(FreeboxOsIconProvider.class);
-    private static final int REQUEST_TIMEOUT_MS = 8000;
 
     private final HttpClient httpClient;
     private final UriBuilder uriBuilder;
+    private final BundleContext context;
 
     @Activate
-    public FreeboxOsIconProvider(final @Reference TranslationProvider i18nProvider,
+    public FreeboxOsIconProvider(final BundleContext context, final @Reference TranslationProvider i18nProvider,
             final @Reference HttpClientFactory httpClientFactory) {
         super(i18nProvider);
+        this.context = context;
         this.httpClient = httpClientFactory.getCommonHttpClient();
         this.uriBuilder = UriBuilder.fromPath("/").scheme("http").host(FreeboxTlsCertificateProvider.DEFAULT_NAME)
                 .path("resources/images/home/pictos");
     }
 
     @Override
     public Set<IconSet> getIconSets(@Nullable Locale locale) {
-        return Set.of();
+        String label = getText("label", DEFAULT_LABEL, locale);
+        String description = getText("decription", DEFAULT_DESCRIPTION, locale);
+
+        return Set.of(new IconSet(BINDING_ID, label, description, Set.of(Format.SVG)));
+    }
+
+    private String getText(String entry, String defaultValue, @Nullable Locale locale) {
+        String text = defaultValue;
+        if (locale != null) {
+            text = i18nProvider.getText(context.getBundle(), "iconset." + entry, defaultValue, locale);
```

Comment:
I believe you can directly call `i18nProvider.getText` without checking first if locale is null.

Reply:
You're right.